js/reference.js

```diff
@@ -5757,6 +5765,19 @@
              return this._linkedData;
          },
 
+          /**
+          * Foreign key data RID names.
+          * During the read we get extra information about the foreign keys,
+          * client could use these extra information for different purposes.
+          * One of these usecases is domain_filter_pattern which they can
+          * include foreignkey data in the pattern language.
+          *
+          * @type {Object}
+          */
+          get linkedDataRIDs() {
```

Comment:
At first I didn't like this new API but now that I saw where you're using this in chaise I think this makes sense. And your comment explains it very well.